src/ext/serializer.c

```diff
@@ -335,15 +336,27 @@ static void _serialize_exception(zval *el, zval *meta, ddtrace_span_t *span TSRM
 }
 
 static void _serialize_meta(zval *el, ddtrace_span_t *span TSRMLS_DC) {
-    zval *meta = _read_span_property(span->span_data, "meta", sizeof("meta") - 1 TSRMLS_CC);
-    BOOL_T should_free = TRUE;
-
-    if (!meta || Z_TYPE_P(meta) != IS_ARRAY) {
-        ALLOC_INIT_ZVAL(meta);
-        array_init(meta);
-    } else {
-        should_free = FALSE;
-        Z_ADDREF_P(meta);
+    zval *meta, *orig_meta = _read_span_property(span->span_data, "meta", sizeof("meta") - 1 TSRMLS_CC);
+    ALLOC_INIT_ZVAL(meta);
+    array_init(meta);
+    if (orig_meta && Z_TYPE_P(orig_meta) == IS_ARRAY) {
+        int key_type;
+        zval **orig_val;
+        zval *val_as_string;
+        HashPosition pos;
+        char *str_key;
+        uint str_key_len;
+        ulong num_key;
+        zend_hash_internal_pointer_reset_ex(Z_ARRVAL_P(orig_meta), &pos);
+        while (zend_hash_get_current_data_ex(Z_ARRVAL_P(orig_meta), (void **)&orig_val, &pos) == SUCCESS) {
+            key_type = zend_hash_get_current_key_ex(Z_ARRVAL_P(orig_meta), &str_key, &str_key_len, &num_key, 0, &pos);
+            if (key_type == HASH_KEY_IS_STRING) {
+                ALLOC_INIT_ZVAL(val_as_string);
+                ddtrace_convert_to_string(val_as_string, *orig_val TSRMLS_CC);
+                add_assoc_zval_ex(meta, str_key, str_key_len, val_as_string);
+            }
+            zend_hash_move_forward_ex(Z_ARRVAL_P(orig_meta), &pos);
+        }
```

Comment:
Was this code copied and modified from anywhere, such as `array_map` or anything?

Reply:
I did look at the `array_*()` functions to remember how to iterate over an array in PHP 5, but I don't recall which ones exactly. Since we're starting to see this show up a lot more in the extension, perhaps we should start using [Sara's array API](https://github.com/sgolemon/php-array-api/blob/master/php_array.h)?